arbeitszeit/use_cases/select_productive_consumption.py

```diff
@@ -0,0 +1,70 @@
+from dataclasses import dataclass
+from uuid import UUID
+
+from arbeitszeit.datetime_service import DatetimeService
+from arbeitszeit.records import ConsumptionType
+from arbeitszeit.repositories import DatabaseGateway
+
+
+@dataclass
+class Request:
+    plan_id: UUID | None
+    amount: int | None
+    consumption_type: ConsumptionType | None
+
+
+@dataclass
+class NoPlanResponse:
+    amount: int | None
+    consumption_type: ConsumptionType | None
+
+
+@dataclass
+class InvalidPlanResponse:
+    amount: int | None
+    consumption_type: ConsumptionType | None
+
+
```

Comment:
All these classes have the same structure.  Wouldn't this be an obvious case for inheritance?

Reply:
It is a software engineering benefit:  it avoids code duplication.  If you wish later to change the basic pattern — say, to add a new field — it is easy to do that in one place.  So it makes your code cleaner.